api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -114,4 +116,23 @@ public Person updateFHIRPerson(Person thePerson, String theId) {
 			return createFHIRPerson(thePerson);
 		}
 	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.PersonService#retirePerson(String)
+	 */
+	@Override
+	public void retirePerson(String id) throws ResourceNotFoundException, NotModifiedException {
+		org.openmrs.Person person = Context.getPersonService().getPersonByUuid(id);
+		if (person == null) throw new ResourceNotFoundException(String.format("Person with id '%s' not found", id));
+		if (person.isPersonVoided()) return;
+		try {
+			person.setVoided(true);
+			person.setPersonVoidedBy(Context.getAuthenticatedUser());
+			person.setPersonVoidReason("Retired by FHIR Request");
+			person.setPersonDateVoided(new Date());
+			Context.getPersonService().voidPerson(person, "Retired by FHIR Request");
+		} catch (APIException apie) {
+			throw new MethodNotAllowedException(String.format("OpenMRS has failed to retire person '%s': %s", id, apie.getMessage()));
```

Comment:
This shouldn't return the type of method not allowed. Can you have a look at existing exceptions which matcch this to this case?